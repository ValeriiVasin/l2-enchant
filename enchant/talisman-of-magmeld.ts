import { createEnchantMap } from '../helpers/create-enchant-map';
import { itemsRange } from '../helpers/items-range';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

// https://l2central.info/classic/items/accessary/bracelet/talisman/#talmagm
export const talismanOfMagmeldEnchantMap = createEnchantMap({
  strategy: EnchantStrategy.Drop,
  enchantItem: Item.MagmeldFragment,
  chances: [
    60, 55, 50, 40, 50, 45, 40, 35, 30, 50, 45, 40, 40, 35, 30, 25, 50, 35, 30, 25, 20, 15, 50, 35, 25, 20, 15, 10, 30,
  ],
  items: itemsRange(Item.TalismanOfMagmeld_1, Item.TalismanOfMagmeld_30),
  boundaries: new Set([
    Item.TalismanOfMagmeld_6,
    Item.TalismanOfMagmeld_12,
    Item.TalismanOfMagmeld_18,
    Item.TalismanOfMagmeld_24,
  ]),
});
