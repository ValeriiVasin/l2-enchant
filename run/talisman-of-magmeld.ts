import { talismanOfMagmeldEnchantMap } from '../enchant/talisman-of-magmeld';
import { itemsRange } from '../helpers/items-range';
import { run } from '../helpers/run';
import { Item } from '../items';

run(
  Item.TalismanOfMagmeld_1,
  itemsRange(Item.TalismanOfMagmeld_2, Item.TalismanOfMagmeld_30),
  100_000,
  talismanOfMagmeldEnchantMap,
);
