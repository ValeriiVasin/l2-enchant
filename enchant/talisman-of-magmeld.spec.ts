import { talismanOfMagmeldEnchantMap } from './talisman-of-magmeld';

it('enchant map: benir', () => {
  expect(talismanOfMagmeldEnchantMap).toMatchSnapshot();
});
