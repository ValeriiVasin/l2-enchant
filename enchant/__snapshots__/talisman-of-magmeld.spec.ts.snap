// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: benir 1`] = `
Map {
  "Талисман Магмельда Ур.1" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.1",
    "results": Array [
      Object {
        "chance": 60,
        "item": "Талисман Магмельда Ур.2",
      },
      Object {
        "chance": 40,
        "item": "Талисман Магмельда Ур.1",
      },
    ],
  },
  "Талисман Магмельда Ур.2" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.2",
    "results": Array [
      Object {
        "chance": 55,
        "item": "Талисман Магмельда Ур.3",
      },
      Object {
        "chance": 45,
        "item": "Талисман Магмельда Ур.1",
      },
    ],
  },
  "Талисман Магмельда Ур.3" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.3",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Магмельда Ур.4",
      },
      Object {
        "chance": 50,
        "item": "Талисман Магмельда Ур.1",
      },
    ],
  },
  "Талисман Магмельда Ур.4" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.4",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Талисман Магмельда Ур.5",
      },
      Object {
        "chance": 60,
        "item": "Талисман Магмельда Ур.1",
      },
    ],
  },
  "Талисман Магмельда Ур.5" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.5",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Магмельда Ур.6",
      },
      Object {
        "chance": 50,
        "item": "Талисман Магмельда Ур.1",
      },
    ],
  },
  "Талисман Магмельда Ур.6" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.6",
    "results": Array [
      Object {
        "chance": 45,
        "item": "Талисман Магмельда Ур.7",
      },
      Object {
        "chance": 55,
        "item": "Талисман Магмельда Ур.6",
      },
    ],
  },
  "Талисман Магмельда Ур.7" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.7",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Талисман Магмельда Ур.8",
      },
      Object {
        "chance": 60,
        "item": "Талисман Магмельда Ур.6",
      },
    ],
  },
  "Талисман Магмельда Ур.8" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.8",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Талисман Магмельда Ур.9",
      },
      Object {
        "chance": 65,
        "item": "Талисман Магмельда Ур.6",
      },
    ],
  },
  "Талисман Магмельда Ур.9" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.9",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Талисман Магмельда Ур.10",
      },
      Object {
        "chance": 70,
        "item": "Талисман Магмельда Ур.6",
      },
    ],
  },
  "Талисман Магмельда Ур.10" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.10",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Магмельда Ур.11",
      },
      Object {
        "chance": 50,
        "item": "Талисман Магмельда Ур.6",
      },
    ],
  },
  "Талисман Магмельда Ур.11" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.11",
    "results": Array [
      Object {
        "chance": 45,
        "item": "Талисман Магмельда Ур.12",
      },
      Object {
        "chance": 55,
        "item": "Талисман Магмельда Ур.6",
      },
    ],
  },
  "Талисман Магмельда Ур.12" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.12",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Талисман Магмельда Ур.13",
      },
      Object {
        "chance": 60,
        "item": "Талисман Магмельда Ур.12",
      },
    ],
  },
  "Талисман Магмельда Ур.13" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.13",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Талисман Магмельда Ур.14",
      },
      Object {
        "chance": 60,
        "item": "Талисман Магмельда Ур.12",
      },
    ],
  },
  "Талисман Магмельда Ур.14" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.14",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Талисман Магмельда Ур.15",
      },
      Object {
        "chance": 65,
        "item": "Талисман Магмельда Ур.12",
      },
    ],
  },
  "Талисман Магмельда Ур.15" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.15",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Талисман Магмельда Ур.16",
      },
      Object {
        "chance": 70,
        "item": "Талисман Магмельда Ур.12",
      },
    ],
  },
  "Талисман Магмельда Ур.16" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.16",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Талисман Магмельда Ур.17",
      },
      Object {
        "chance": 75,
        "item": "Талисман Магмельда Ур.12",
      },
    ],
  },
  "Талисман Магмельда Ур.17" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.17",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Магмельда Ур.18",
      },
      Object {
        "chance": 50,
        "item": "Талисман Магмельда Ур.12",
      },
    ],
  },
  "Талисман Магмельда Ур.18" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.18",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Талисман Магмельда Ур.19",
      },
      Object {
        "chance": 65,
        "item": "Талисман Магмельда Ур.18",
      },
    ],
  },
  "Талисман Магмельда Ур.19" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.19",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Талисман Магмельда Ур.20",
      },
      Object {
        "chance": 70,
        "item": "Талисман Магмельда Ур.18",
      },
    ],
  },
  "Талисман Магмельда Ур.20" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.20",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Талисман Магмельда Ур.21",
      },
      Object {
        "chance": 75,
        "item": "Талисман Магмельда Ур.18",
      },
    ],
  },
  "Талисман Магмельда Ур.21" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.21",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Талисман Магмельда Ур.22",
      },
      Object {
        "chance": 80,
        "item": "Талисман Магмельда Ур.18",
      },
    ],
  },
  "Талисман Магмельда Ур.22" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.22",
    "results": Array [
      Object {
        "chance": 15,
        "item": "Талисман Магмельда Ур.23",
      },
      Object {
        "chance": 85,
        "item": "Талисман Магмельда Ур.18",
      },
    ],
  },
  "Талисман Магмельда Ур.23" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.23",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Магмельда Ур.24",
      },
      Object {
        "chance": 50,
        "item": "Талисман Магмельда Ур.18",
      },
    ],
  },
  "Талисман Магмельда Ур.24" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.24",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Талисман Магмельда Ур.25",
      },
      Object {
        "chance": 65,
        "item": "Талисман Магмельда Ур.24",
      },
    ],
  },
  "Талисман Магмельда Ур.25" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.25",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Талисман Магмельда Ур.26",
      },
      Object {
        "chance": 75,
        "item": "Талисман Магмельда Ур.24",
      },
    ],
  },
  "Талисман Магмельда Ур.26" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.26",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Талисман Магмельда Ур.27",
      },
      Object {
        "chance": 80,
        "item": "Талисман Магмельда Ур.24",
      },
    ],
  },
  "Талисман Магмельда Ур.27" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.27",
    "results": Array [
      Object {
        "chance": 15,
        "item": "Талисман Магмельда Ур.28",
      },
      Object {
        "chance": 85,
        "item": "Талисман Магмельда Ур.24",
      },
    ],
  },
  "Талисман Магмельда Ур.28" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.28",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Талисман Магмельда Ур.29",
      },
      Object {
        "chance": 90,
        "item": "Талисман Магмельда Ур.24",
      },
    ],
  },
  "Талисман Магмельда Ур.29" => Object {
    "cost": "Осколок Талисмана Магмельда",
    "item": "Талисман Магмельда Ур.29",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Талисман Магмельда Ур.30",
      },
      Object {
        "chance": 70,
        "item": "Талисман Магмельда Ур.24",
      },
    ],
  },
}
`;
